tripal_chado/src/Controller/ChadoOrganismFormElementController.php

```diff
@@ -0,0 +1,342 @@
+      $element = self::getSelectElement($element, $default, $options);
+    }
+    return $element;
+  }
+
+  /**
+   * The select form element.
+   *
+   * @param array $element
+   *   The form element array to be populated.
+   * @param mixed $default
+   *   The default value, either an integer pkey ID or a string.
+   * @param array $options
+   *   select_limit - The maximum number of options to show in a select list.
+   *   match_operator - Either 'CONTAINS' (default) or 'STARTS_WITH'.
+   *   match_limit - Desired number of matching names to suggest.
+   *
+   * @return array|null
+   *   A form element array, either a select element.
+   */
+  public static function getSelectElement(array $element, mixed $default, array $options): ?array {
+    $select_options = self::getSelectOptions($options);
+    natcasesort($select_options);
+    $default_id = 0;
+    if (gettype($default) == 'integer') {
+      $default_id = $default;
+    }
+    elseif (gettype($default) == 'string') {
+      $default_id = self::getPkeyId($default);
+    }
+    $element = [
+      '#type' => 'select',
+      '#options' => $select_options,
+      '#default_value' => $default_id,
+      '#empty_option' => t('- Select -'),
+    ];
+    $element['#element_validate'] = [[static::class, 'validateAutocomplete']];
+    return $element;
+  }
+
+  /**
+   * The autocomplete form element.
+   *
+   * @param array $element
+   *   The form element array to be populated.
+   * @param mixed $default
+   *   The default value, either an integer pkey ID or a string.
+   * @param array $options
+   *   select_limit - The maximum number of options to show in a select list.
+   *   match_operator - Either 'CONTAINS' (default) or 'STARTS_WITH'.
+   *   match_limit - Desired number of autocomplete matching names to suggest.
+   *   size - The size of the textfield for autocomplete.
+   *   placeholder - Placeholder text for the autocomplete textfield.
+   *
+   * @return array|null
+   *   A form element array, an autocomplete.
+   */
+  public static function getAutocompleteElement(array $element, mixed $default, array $options): ?array {
+    if (gettype($default) == 'string') {
+      $default_value = $default;
+    }
+    elseif (gettype($default) == 'integer') {
+      $default_id = $default;
+      $default_value = '';
+      if ($default_id) {
+        // Construct a query
+        // A single wildcard indicates that all records are to be returned.
+        $string = '%';
+        // Add one to select limit so we know if it is exceeded.
+        $count_options = $options;
+        $count_options['match_limit'] = $options['select_limit'] + 1;
+        $query = self::getQuery($string, $count_options);
+        // We can reuse the existing query since only one change is needed.
+        $query->condition('organism_id', $default_id, '=');
+        $result = $query->execute()->fetchObject();
+        if ($result) {
+          // Strip HTML tags if present, e.g. in Pub title.
```

Comment:
```suggestion
          // Strip HTML tags if present, but this is unlikely for organism.
```
Use the same comment you have earlier.